src/states_screens/ghost_replay_selection.cpp

```diff
@@ -97,36 +139,193 @@ void GhostReplaySelection::loadList()
     {
+        if (m_best_times)
+        {
+            // All best times have already been added
+            if (best_index + 1 > m_best_times_index.size())
+                break;
+            // This works because the best times index is already sorted
+            else if (m_best_times_index[best_index] == i)
+                best_index++;
+            // There are still best times to display
+            // The current i don't correspond to a best time
+            else
+                continue;
+        }
+
         const ReplayPlay::ReplayData& rd = ReplayPlay::get()->getReplayData(i);
 
         if (m_same_difficulty && m_cur_difficulty !=
             (RaceManager::Difficulty)rd.m_difficulty)
             continue;
 
+        core::stringw current_version = STK_VERSION;
+        if (m_same_version && current_version != rd.m_stk_version)
+            continue;
+
+        // Only display replays comparable with the replay selected for comparison
+        if (m_is_comparing)
+        {
+                // If it's not the same track, check further in the index
+                if (rd.m_track_name != rd_compare.m_track_name)
+                    continue;
+
+                // If it's not the same direction, check further in the index
+                if (rd.m_reverse != rd_compare.m_reverse)
+                    continue;
+
+                // If it's not the same lap numbers, check further in the index
+                if (rd.m_laps != rd_compare.m_laps)
+                    continue;
+
+                // Don't compare a replay with itself
+                if (compare_index == i)
+                    continue;
+        }
+        // Only display replays matching the current mode
+        if (m_active_mode == RaceManager::MINOR_MODE_TIME_TRIAL &&
+            rd.m_minor_mode != "time-trial")
+            continue;
+        else if (m_active_mode == RaceManager::MINOR_MODE_EASTER_EGG &&
+            rd.m_minor_mode != "egg-hunt")
+            continue;
+
         Track* track = track_manager->getTrack(rd.m_track_name);
         
         if (track == NULL)
             continue;
 
         std::vector<GUIEngine::ListWidget::ListCell> row;
+        //The third argument should match the numbers used in beforeAddingWidget
         row.push_back(GUIEngine::ListWidget::ListCell
-            (translations->fribidize(track->getName()) , -1, 3));
+            (translations->fribidize(track->getName()) , -1, 9));
         row.push_back(GUIEngine::ListWidget::ListCell
-            (StringUtils::toWString(rd.m_kart_list.size()), -1, 1, true));
+            (StringUtils::toWString(rd.m_kart_list.size()), -1, 3, true));
+        if (m_active_mode_is_linear)
+            row.push_back(GUIEngine::ListWidget::ListCell
+                (rd.m_reverse ? _("Yes") : _("No"), -1, 3, true));
+        if (!m_same_difficulty)
+            row.push_back(GUIEngine::ListWidget::ListCell
+                (rd.m_difficulty == 3 ? _("SuperTux") : rd.m_difficulty == 2 ?
```

Comment:
RaceManager::getDifficultyName could be reused instead of hardcoding this here (I know this isn't exactly your code, but if the code is being changed we might as well improve it)